app/src/main/java/com/example/secret/model/FirebaseModel.java

```diff
@@ -139,6 +139,47 @@ public void signOut() {
         auth.signOut();
     }
 
+    public void getComment(String commentId, Listener<Map<String, Object>> successListener, Listener<Void> failedListener) {
+        db.collection(Comment.COLLECTION).document(commentId).get()
+                .addOnSuccessListener(t -> {
+                    if (t.exists()) {
+                        successListener.onComplete(t.getData());
+                        return;
+                    }
+                    failedListener.onComplete(null);
+                })
+                .addOnFailureListener(t -> failedListener.onComplete(null));
+    }
+
+    public void setComment(Comment comment, Listener<Void> successListener, Listener<Void> failListener) {
+        db.collection(Post.COLLECTION).document(comment.getId()).set(comment.toJson())
+                .addOnCompleteListener(task -> {
+                    if (task.isSuccessful()) {
+                        successListener.onComplete(null);
+                        return;
+                    }
+                    Log.e("SetComment", "Could not set Comment");
+                    failListener.onComplete(null);
+                });
+    }
+
+    public void getAllCommentsSince(Long since, Listener<List<Comment>> callback) {
```

Comment:
I think you don't need the since.
This is optional, but we only have 2 states - show 2 latest comments and show all comments.
So the since is unnecessary. Correct me if Im wrong

Reply:
You are wrong
We only want to fetch the delta from firebase
The queries is only to the localDB